controller/volume_controller.go

```diff
@@ -1091,6 +1097,24 @@ func (vc *VolumeController) ReconcileVolumeState(v *longhorn.Volume, es map[stri
 				scheduled = true
 			}
 		}
+
+		failureMessage := ""
+		if !scheduled {
+			if len(aggregatedVerboseErrors) == 0 {
+				aggregatedVerboseErrors.AppendError(longhorn.SchedulerErrorSchedulingFailed)
+			}
+			failureMessage = aggregatedVerboseErrors.Join()
+			scheduledCondition := types.GetCondition(v.Status.Conditions, longhorn.VolumeConditionTypeScheduled)
+			if scheduledCondition.Status == longhorn.ConditionStatusFalse {
+				v.Status.Conditions = types.SetCondition(v.Status.Conditions,
+					longhorn.VolumeConditionTypeScheduled, longhorn.ConditionStatusFalse,
+					scheduledCondition.Reason,
+					failureMessage)
+			}
+		}
+		if err := vc.updatePVAnnotation(v, types.PVAnnotationLonghornVolumeSchedulingError, failureMessage); err != nil {
+			log.Warnf("Cannot update PV annotation for volume %v", v.Name)
```

Comment:
This is good. We should just warn out instead of returning the error, because this is an optional update outside Longhorn resources.